Support receiving sparse histograms (#2492)
grafana/mimir: docs/sources/operators-guide/mimir-runbooks/_index.md

### err-mimir-too-far-in-future

> This non-critical error occurs when Mimir receives a write request that contains a sample or histogram whose timestamp is in the future compared to the current "real world" time.

Reviewer: ```suggestion
This is a non-critical error. It occurs when Mimir receives a write request that contains a sample or histogram whose timestamp is in the future compared to the current _real world_ time.
```
What does "real world" time mean? Wall clock time?

- Ensure each compactor replica has successfully updated bucket index of each owned tenant within the double of `-compactor.cleanup-interval` (query below assumes the cleanup interval is set to 15 minutes):
  `time() - cortex_compactor_block_cleanup_last_successful_run_timestamp_seconds > 2 * (15 * 60)`

### err-mimir-histogram-different-number-spans-buckets

This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose either negative or positive spans specify
a different number of buckets then there are provided.

> > **Note**: Series with invalid histograms are skipped during the ingestion, and series within the same request are ingested.

Reviewer 2: I did not understand the `series within the same request are ingested` part

Author: I took it from one of the other errors, but I presume it means: "other series that don't have invalid samples/histograms that are part of the same request are still ingested"

- Ensure each compactor replica has successfully updated bucket index of each owned tenant within the double of `-compactor.cleanup-interval` (query below assumes the cleanup interval is set to 15 minutes):
  `time() - cortex_compactor_block_cleanup_last_successful_run_timestamp_seconds > 2 * (15 * 60)`

### err-mimir-histogram-spans-buckets-mismatch

This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose either negative or positive spans specify

> a different number of buckets than there are provided.

Reviewer: ```suggestion
a different number of buckets than those that are provided.
```

> This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose either negative or positive spans specify
> a different number of buckets than there are provided.

Reviewer: ```suggestion
This is a non-critical error. It occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose (either negative or positive) spans specify
a different number of buckets than are provided.
```

Reviewer: This sentence is pretty complex, so I attempted to simplify the phrases.

> > **Note**: Series with invalid histograms are skipped during the ingestion, and series within the same request are ingested.

Reviewer: ```suggestion
> **Note:** Series with invalid histograms are skipped during the ingestion, and series within the same request are ingested.
```

Reviewer: Please change throughout.

### err-mimir-histogram-span-negative-offset

> This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram which has a span whose offset is negative, making it overlap with the previous span.

Reviewer: ```suggestion
This is a non-critical error. It occurs when Mimir receives a write request that contains a native (also known as sparse) histogram that has a span whose offset is negative, which makes it overlap with the previous span.
```

> **Note**: Series with invalid histograms are skipped during the ingestion, and series within the same request are ingested.

### err-mimir-histogram-negative-bucket-count

> This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram which has a bucket whose observation count is negative.

Reviewer: ```suggestion
This is a non-critical error. It occurs when Mimir receives a write request that contains a native (also known as sparse) histogram that has a bucket whose observation count is negative.
```

> **Note**: Series with invalid histograms are skipped during the ingestion, and series within the same request are ingested.

### err-mimir-histogram-count-not-big-enough

> This non-critical error occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose overall observation count is lower than the number of observations found in the buckets.

Reviewer: ```suggestion
This is a non-critical error. It occurs when Mimir receives a write request that contains a native (also known as sparse) histogram whose overall observation count is lower than the number of observations found in the buckets.
```
This is a very long sentence. Is there a way to break down its complexity for non-native English speakers who have to store a lot of grammar in (human) memory?

Reviewer: What is an "overall" observation count? Can we remove this term?
